src/dscord/bot/command.d

```diff
@@ -1,28 +1,80 @@
 /**
-  Utilties for building user-controlled commands with the dscord bot interface
+  Utilities for building user-controlled commands with the dscord bot interface
 */
 
 module dscord.bot.command;
 
 import std.regex,
        std.array,
-       std.functional,
-       std.algorithm;
+       std.algorithm,
+       std.string;
 
 import dscord.types,
        dscord.gateway;
 
-static struct CommandDefinition {
-  string[]  triggers;
+/// Commands are members of Plugin class or classes which inherit from it. See: examples/src/basic.d
+/// Example usage:
+///
+/// @Command("hello", "hello2")    //The command may respond to multiple triggers.
+/// @Enabled(true)                 //false turns the command off. It may be re-enabled elsewhere in code at run-time.
+/// @Description("A command that says 'hello' in channel.")    //A description of what the command does.
+/// @Group("Leet")                 //The group this command can respond to.
+/// @RegEx(true)                   //Defaults to true, but can disable RegEx handling for command triggers.
+/// @CommandLevel(1)               //The power level that the command requires ( normal = 1, mod = 50, admin = 100,)
+/// void onHello(CommandEvent event) {
+///    event.msg.reply("Hello, world!");
+/// }
+//Easy aliases to use for UDAs
+//A UDA that can be used to flag a function as a command handler in a Plugin.
+alias Command = CommandConfig!TypeTriggers;
+alias Enabled = CommandConfig!TypeEnabled;
+alias Description = CommandConfig!TypeDescription;
+alias Group = CommandConfig!TypeGroup;
+alias RegEx = CommandConfig!TypeRegEx;
+alias CommandLevel = CommandConfig!TypeLevel;
+
+//Aliases for deprecated UDAs
+alias CommandDescription = CommandConfig!TypeDescription;
+alias CommandGroup = CommandConfig!TypeGroup;
+alias CommandRegex = CommandConfig!TypeRegEx;
+
+//Custom types to select which CommandConfig details we get.
+//Note: The names *must* correspond to property names in CommandObject.
+enum TypeEnabled = "enabled";
+enum TypeTriggers = "triggers";
+enum TypeDescription = "description";
+enum TypeRegEx = "useRegex";
+enum TypeGroup = "group";
+enum TypeLevel = "level";
+
+//Template overloads for implementations of each type of CommandConfig info
+template CommandConfig(alias T) if(T==TypeTriggers){
```

Comment:
for these I'd go with `if (T == TypeTriggers) {`, I tend to break std-D style a bit but I still prefer it.